crates/ide-assists/src/utils.rs

```diff
@@ -316,43 +316,52 @@ pub(crate) fn does_pat_match_variant(pat: &ast::Pat, var: &ast::Pat) -> bool {
     pat_head == var_head
 }
 
-pub(crate) fn does_nested_pattern(pat: &ast::Pat) -> bool {
-    let depth = calc_depth(pat, 0);
+pub(crate) fn does_pat_variant_nested_or_literal(pat: &ast::Pat) -> bool {
+    check_pat_variant_nested_or_literal_with_depth(pat, 0)
+}
 
-    if 1 < depth {
+fn check_pat_variant_nested_or_literal_with_depth(pat: &ast::Pat, depth: usize) -> bool {
+    if depth > 1 {
         return true;
     }
-    false
-}
 
-fn calc_depth(pat: &ast::Pat, depth: usize) -> usize {
     match pat {
         ast::Pat::IdentPat(_)
-        | ast::Pat::BoxPat(_)
         | ast::Pat::RestPat(_)
-        | ast::Pat::LiteralPat(_)
-        | ast::Pat::MacroPat(_)
-        | ast::Pat::OrPat(_)
-        | ast::Pat::ParenPat(_)
-        | ast::Pat::PathPat(_)
         | ast::Pat::WildcardPat(_)
-        | ast::Pat::RangePat(_)
-        | ast::Pat::RecordPat(_)
         | ast::Pat::RefPat(_)
-        | ast::Pat::SlicePat(_)
-        | ast::Pat::TuplePat(_)
-        | ast::Pat::ConstBlockPat(_) => depth,
+        | ast::Pat::BoxPat(_)
+        | ast::Pat::ConstBlockPat(_) => false,
```

Comment:
Const block pattern is comparing to a const (similar to a literal) - so it should be `true` (there is an exception for unit-struct consts but it's hard to check).